skxray/core/roi.py

```diff
@@ -357,39 +356,35 @@ def roi_pixel_values(image, labels, index=None):
     for n in index:
         roi_pix.append(image[labels == n])
     return roi_pix, index
+    
 
-
-def mean_intensity_sets(images_set, labels):
-    """
-    Mean intensities for ROIS' of the labeled array for different image sets
-
+def mean_intensity_sets(data_dict, labeled_array):
+    """Create a dataframe with columns as the data sets and rows as the ROIs
+    
     Parameters
     ----------
-    images_set : array
-        images sets
-        shapes is: (len(images_sets), )
-        one images_set is iterable of 2D arrays dimensions are: (rr, cc)
-
-    labels : array
+    data_dict : dict
+        Dictionary of 2-D image stacks. keys are data set names, values are
+        2-D image stacks
+    labeled_array : array
         labeled array; 0 is background.
         Each ROI is represented by a distinct label (i.e., integer).
 
     Returns
     -------
-    mean_intensity_list : list
-        average intensity of each ROI as a list
-        shape len(images_sets)
-
-    index_list : list
-        labels list for each image set
-
+    dataframe : pd.DataFrame
+        Pandas dataframe where the columns are the data sets and the rows are
+        the 1-D roi's
     """
-    return tuple(map(list,
-                     zip(*[mean_intensity(im,
-                                          labels) for im in images_set])))
-
-
-def mean_intensity(images, labels, index=None):
+    series_dict = {}
+    for name, data in data_dict.items():
+        by_roi, roi_list = _mean_intensity(data, labeled_array)
+        data = [by_roi[:, i] for i in range(by_roi.shape[1])]
+        series_dict[name] = pd.Series(data=data, index=['roi%s' % idx for idx in roi_list])
+    return pd.DataFrame(series_dict)
+    
+
+def _mean_intensity(images, labeled_array, index=None):
```

Comment:
@ericdill I saw you changed function name, you have to change that in test too, because I put a test for that function